packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart

```diff
@@ -136,6 +143,69 @@ final class AssetTransformer {
   }
 }
 
+
+// A wrapper around [AssetTransformer] to support hot reload of transformed assets.
+final class DevelopmentAssetTransformer {
+  DevelopmentAssetTransformer({
+    required FileSystem fileSystem,
+    required AssetTransformer transformer,
+    required Logger logger,
+  })  : _fileSystem = fileSystem,
+        _transformer = transformer,
+        _logger = logger;
+
+  final AssetTransformer _transformer;
+  final FileSystem _fileSystem;
+  final Pool _transformationPool = Pool(4);
+  final Logger _logger;
+
+  /// Re-transforms an asset and returns a [DevFSContent] that should be synced
+  /// to the attached device in its place.
+  ///
+  /// Returns `null` if any of the transformation subprocesses failed.
+  Future<DevFSContent?> retransformAsset({
+    required String inputAssetKey,
+    required DevFSContent inputAssetContent,
+    required List<AssetTransformerEntry> transformerEntries,
+    required String workingDirectory,
+  }) async {
+    final File output = _fileSystem.systemTempDirectory.childFile('retransformerInput-$inputAssetKey');
+    ErrorHandlingFileSystem.deleteIfExists(output);
+    File? inputFile;
+    bool cleanupInput = false;
+    Uint8List result;
+    PoolResource? resource;
+    try {
+      resource = await _transformationPool.request();
+      if (inputAssetContent is DevFSFileContent) {
+        inputFile = inputAssetContent.file as File;
+      } else {
+        inputFile = _fileSystem.systemTempDirectory.childFile('retransformerInput-$inputAssetKey');
+        inputFile.writeAsBytesSync(await inputAssetContent.contentsAsBytes());
+        cleanupInput = true;
+      }
+      final AssetTransformationFailure? failure = await _transformer.transformAsset(
+        asset: inputFile,
+        outputPath: output.path,
+        transformerEntries: transformerEntries,
+        workingDirectory: workingDirectory,
+      );
+      if (failure != null) {
+        _logger.printError(failure.message);
+        return null;
+      }
+      result = output.readAsBytesSync();
+    } finally {
+      resource?.release();
+      ErrorHandlingFileSystem.deleteIfExists(output);
+      if (cleanupInput) {
+        ErrorHandlingFileSystem.deleteIfExists(inputFile!);
```

Comment:
it could be null line 179 `resource = await _transformationPool.request();` throws, right?

Reply:
I've added a null check for extra safety.